### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by MesakDuduCoder**
## In this branch I implemented the following

* Create a class Classroom, Book and Rental
* Create the has-many/belongs-to relationship between Classroom and Student
* Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+
 class Student < Person
-  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
-    super(age, parent_permission, name)
+  attr_accessor :name, :classroom
+
+  def initialize(age, classroom, name: 'Unknown')
+    super(age, name: name)
     @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```
Comment (youmari) [inline student.rb:9]: - [ ] Nice adding student to the classroom automatically but it should be in the classroom method like the following 
```suggestion
    def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end

```
that way your student can have many classrooms and the same for a classroom can have many students
and also will not return an error when there is no classroom

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @MesakDuduCoder , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (MesakDuduCoder) [reply student.rb:14]: Okay I have done it

**Review by mukhammadrizooff — APPROVED** [review]
Hi @MesakDuduCoder, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**